packages/ods/src/components/timepicker/src/components/ods-timepicker/ods-timepicker.tsx

```diff
@@ -0,0 +1,174 @@
+import type { OdsTimepickerValueChangeEventDetail } from '../../interfaces/event';
+import type { EventEmitter, FunctionalComponent } from '@stencil/core';
+import { AttachInternals, Component, Element, Event, Host, Method, Prop, Watch, h } from '@stencil/core';
+import { ODS_INPUT_TYPE, type OdsInput, type OdsInputValueChangeEvent } from '../../../../input/src';
+import { type OdsSelectEventChange } from '../../../../select/src';
+import { type OdsTimezonePreset } from '../../constant/timezone-preset';
+import { type ODS_TIMEZONE, type OdsTimezone } from '../../constant/timezones';
+import { formatValue, getCurrentTimezone, parseTimezones, setFormValue } from '../../controller/ods-timepicker';
+
+@Component({
+  formAssociated: true,
+  shadow: true,
+  styleUrl: 'ods-timepicker.scss',
+  tag: 'ods-timepicker',
+})
+export class OdsTimepicker {
+  private defaultCurrentTimezone?: OdsTimezone;
+  private odsInput?: OdsInput & HTMLElement;
+  private odsSelect?: OdsSelect & HTMLElement;
+  private timezonesList: ODS_TIMEZONE[] = [];
+
+  @Element() el!: HTMLElement;
+
+  @AttachInternals() private internals!: ElementInternals;
+
+  @Prop({ reflect: true }) public ariaLabel: HTMLElement['ariaLabel'] = null;
+  @Prop({ reflect: true }) public ariaLabelledby?: string;
+  /** @docType OdsTimezone */
+  @Prop({ mutable: true, reflect: true }) public currentTimezone?: OdsTimezone;
+  @Prop({ reflect: true }) public defaultValue?: string;
+  @Prop({ reflect: true }) public hasError: boolean = false;
+  @Prop({ reflect: true }) public isDisabled: boolean = false;
+  @Prop({ reflect: true }) public isReadonly: boolean = false;
+  @Prop({ reflect: true }) public isRequired: boolean = false;
+  @Prop({ reflect: true }) public name!: string;
+  /** @docType  OdsTimezone[] | OdsTimezonePreset | string */
+  @Prop({ reflect: true }) public timezones?: OdsTimezone[] | OdsTimezonePreset | string;
+  @Prop({ mutable: true, reflect: true }) public value: string | null = null;
+  @Prop({ reflect: true }) public withSeconds: boolean = false;
+
+  @Event() odsBlur!: EventEmitter<void>;
+  @Event() odsChange!: EventEmitter<OdsTimepickerValueChangeEventDetail>;
+  @Event() odsClear!: EventEmitter<void>;
+  @Event() odsFocus!: EventEmitter<void>;
+  @Event() odsReset!: EventEmitter<void>;
+
+  @Method()
+  async clear(): Promise<void> {
+    this.odsInput?.clear();
+    this.odsSelect?.clear();
+    this.odsClear.emit();
+  }
+
+  @Method()
+  async getValidity(): Promise<ValidityState | undefined> {
+    return this.odsInput?.getValidity();
+  }
+
+  @Method()
+  async reset(): Promise<void> {
+    this.odsInput?.reset();
+    this.odsSelect?.reset();
+    this.odsReset.emit();
+  }
+
+  @Watch('timezones')
+  handleTimezones(): void {
+    this.initTimezones();
+  }
+
+  @Watch('withSeconds')
+  @Watch('value')
+  onValueChange(): void {
+    const value = formatValue(this.odsInput?.value as string, this.withSeconds);
```

Comment:
why the `as string`?
update formatValue to accept the same as the expecteed value Prop (`string | null`) instead

Reply:
In the odsInput the value is `string | number | null`

Reply:
my bad I though this was the local value prop